File: leetcode/136/c++/solution.cpp
/*
    136. Single Number
    Author: Miguel Angel Bermeo Ayerbe
*/
class Solution {
   public:
    int singleNumber(vector<int>& nums) {
        int result = 0;
        for (const int& num_i : nums) {
            result = result ^ num_i;
        }
        return result;
    }
};

File: leetcode/136/rust/solution.rs
/*
    136. Single Number
    Author: Miguel Angel Bermeo Ayerbe
*/
impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut result: i32 = 0;
        for num_i in nums.iter() {
            result ^=*num_i; 
        }
        result
    }
}